contracts/Oxo.sol

```diff
@@ -0,0 +1,575 @@
+        }
+        if (movesCount % 2 != 0) {
+            require(players[msg.sender] == 1);
+            moveTime1 = now;
+        }
+
+        require(field[_col][_row] == 0);
+        field[_col][_row] = players[msg.sender];
+
+        Move(_col, _row, "move success");
+        if (movesCount > 4)checkWiner(0);
+        if (movesCount == 9)checkWiner(9);
+        if (gameOn) movesCount++;
+    }
+
+    function checkWiner(uint _gameOn) internal {
+        // horizontal
+
+        if(field[0][0] == 1 && field[0][1] == 1 && field[0][2] == 1) player1Win();
+        if(field[0][0] == 2 && field[0][1] == 2 && field[0][2] == 2) player2Win();
+
+        if(field[1][0] == 1 && field[1][1] == 1 && field[1][2] == 1) player1Win();
+        if(field[1][0] == 2 && field[1][1] == 2 && field[1][2] == 2) player2Win();
+
+        if(field[2][0] == 1 && field[2][1] == 1 && field[2][2] == 1) player1Win();
+        if(field[2][0] == 2 && field[2][1] == 2 && field[2][2] == 2) player2Win();
+
+        // vertical
+
+        if(field[0][0] == 1 && field[1][0] == 1 && field[2][0] == 1) player1Win();
+        if(field[0][0] == 2 && field[1][0] == 2 && field[2][0] == 2) player2Win();
+
+        if(field[0][1] == 1 && field[1][1] == 1 && field[2][1] == 1) player1Win();
+        if(field[0][1] == 2 && field[1][1] == 2 && field[2][1] == 2) player2Win();
+
+        if(field[0][2] == 1 && field[1][2] == 1 && field[2][2] == 1) player1Win();
+        if(field[0][2] == 2 && field[1][2] == 2 && field[2][2] == 2) player2Win();
+
+        // diagonal
+
+        if(field[0][0] == 1 && field[1][1] == 1 && field[2][2] == 1) player1Win();
+        if(field[0][0] == 2 && field[1][1] == 2 && field[2][2] == 2) player2Win();
+
+        if(field[2][0] == 1 && field[1][1] == 1 && field[0][2] == 1) player1Win();
+        if(field[2][0] == 2 && field[1][1] == 2 && field[0][2] == 2) player2Win();
+
+        // dead heat
+
+        if(_gameOn == 9) deadHeat();
+
+//        Debug('checkWiner end');
+    }
+
+    function stopChallenge()public{
+        // require(msg.sender==owner || msg.sender==player1Addr);
+
+        // Any can stop
+        require(now > timeStart1 + 10 seconds); //10 seconds \ 3 minutes
+        require(playerNum == 2);
+        player1Addr.transfer(bet);
+        bet = 0;
+        gameOn = false;
+        playerNum = 1;
+
+
+        StopChall(player1Addr, players[msg.sender], bet, "bet returned" );
+    }
+
+    function stopWaitingMove() public {
+        require(playerNum == 3);
+
+        if (movesCount == 1){
+            require(now > timeStart2 + 10 seconds);  // 3 minutes
+            player2Win();
+        }
+
+        if (movesCount % 2 == 0){
+            require(now > moveTime1 + 10 seconds);
+            player1Win();
+        }
+
+        if (movesCount % 2 != 0 && movesCount != 1 ){
+            require(now > moveTime2 + 10 seconds);
+            player2Win();
+        }
+
+    }
+
+    function deadHeat() internal {
+        player1Addr.transfer(bet);
```

Comment:
Can you imagine a situation when player2 won't have his money? Like if first player will be a contract instead of human?